### PR TITLE
fix(CardBrowser): retain selection when toggling 'mark'

### LINKED ISSUE #14950: [Bug] Card Browser: "Toggle Mark" removes selection
###### Reproduction Steps

1. Open Card Browser
2. Select a row
3. "Toggle Mark"

###### Expected Result

* row is marked and remains selected

###### Actual Result

* row is deselected

###### Debug info
AnkiDroid Version = 2.17alpha9-debug (ad6e8f40cd3841d4bf58c2f3163a85d1444da66e)

Backend Version = 0.1.30-anki23.10.1

Android Version = 13 (SDK 33)

ProductFlavor = amazon

Manufacturer = Google

Model = sdk_gphone64_arm64

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 13; sdk_gphone64_arm64 Build/TE1A.220922.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/103.0.5060.71 Mobile Safari/537.36

ACRA UUID = febb1791-fb3c-4293-ac53-3220baabc1fa

Crash Reports Enabled = false

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Fixes
* Fixes #14950

## Approach
* allow ViewModel operations to be handled in CardBrowser
* reload the individual cards so `forceRefreshSearch` is not required
* then just a call to `cardsAdapter.notifyDataSetChanged()`

## How Has This Been Tested?
Unit Tested

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -658,7 +658,8 @@ open class CardBrowser :
      * If one or more card is unmarked, all will be marked,
      * otherwise, they will be unmarked  */
     @NeedsTest("Test that the mark get toggled as expected for a list of selected cards")
-    private fun toggleMark() {
+    @VisibleForTesting
+    fun toggleMark() = launchCatchingTask {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:662]: This is reverted in the next commit but why wrap the original coroutine inside another coroutine?

**Review by lukstbit — APPROVED** [review]
LGTM, nice!

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:662]: Thanks! Bad merge fix. 

I added the test afterwards then reordered the commits

**Review by mikehardy — APPROVED** [review]
💥

**Pull request merged**